### ISSUE TITLE
Style override for TextInput

### ISSUE THREAD

**Issue by dominictwlee** [callstack/react-native-paper#1115]
## Describe the feature
Passing down style props to the NativeTextInput component within TextInput. Something like inputStyles that would override the default styles.

https://github.com/callstack/react-native-paper/blob/635d647f46347a5cf014cc1ef9bbb5fc4f9eed98/src/components/TextInput/TextInputOutlined.js#L221-L233

Maybe put it at the end of the styles array here?

## Motivation
I'd like to have full control and override styles if/when needed. For example, right now there isn't a feasible way to override the input text color or font weight when the input is disabled.

**Comment by ferrannp** [comment]
@jbinda what do you think?

**Comment by jbinda** [comment]
1. We already have a ticket reffers to set up TextInput colors (#688) so after I deal with that it should solve one request from @dominictwlee motivation
2. We already implemented feature to set up font size, so we can can also make small adjustment to set up font weight.

I would be careful to give full control to override styles because of the label animation sitting there. After we deal with more flexible way to style colors we should be fine.

Is there anything else you would like to override @dominictwlee ?

**Comment by dominictwlee** [comment]
Yeah I think in the meantime that would do, I'll let you know if I find other use cases for style customization.

**Comment by davesag** [comment]
For the app I am working on I need some of my inputs to use the font `monospace` (for Android) and `Courier New` (for iOS). This should not affect the label, only the actual input itself. The suggestion of adding an `inputStyle` prop is a good one I think and would certainly work for me.

**Comment by solshuffle** [comment]
It would be really useful to have control over the padding and/or height of the `<TextInput />`

**Comment by jbinda** [comment]
@solshuffle you can pass height to style to change TextInput height or dense prop to adjust size to dense mode according to material ui design guidelines

@davesag I will check what we can do and back with answer/solution

I have 2 PR almost done with the listed item I posted before. Will raise it soon

**Comment by solshuffle** [comment]
The `dense` prop doesn't seem to be working and if I try to adjust the height of the `<TextInput />`, the label text inside is not centred, even if I add `textAlignVertical: 'center'` or `justifyContent: 'center'`. 
The padding would be really useful, for example in case we have a `<TextInput />` with a transparent background, and we want to remove the horizontal padding.
Here is an example of what it looks like.

<img width="390" alt="image" src="https://user-images.githubusercontent.com/13484862/61883183-3a9dfc00-aefa-11e9-8ef0-aa7017a85e41.png">

**Comment by jbinda** [comment]
Hi @solshuffle 

Which version of react-native-paper are you using ? Because all of you wrote should be already implemented in 3.0 branch

There is also padding="none" prop that allows to remove horizontal padding as you mentioned.

Please check the below screen from the example app:
![image](https://user-images.githubusercontent.com/21242757/62203897-5edf5a00-b38c-11e9-8fb0-8aa9ac3dfc47.png)

**Comment by solshuffle** [comment]
Thanks @jbinda 
I just upgraded to "3.0.0-alpha.2" and that fixed the padding and dense issues. 
The problem I am having now is that it doesn't recognise the fonts from the theme anymore. 

I get this warning:
<img width="390" alt="image" src="https://user-images.githubusercontent.com/13484862/62223571-74b54500-b3b5-11e9-981e-f767d0506c8d.png">

Here is the code in theme, it used to work and show the intended "Montserrat-Medium", now it shows the default font.

```
import { DefaultTheme } from 'react-native-paper';

export default {
  ...DefaultTheme,
  dark: false,
  roundness: 3,
  fonts: {
    thin: 'Montserrat-Light',
    light: 'Montserrat-Medium',
    regular: 'Montserrat-Medium',
    medium: 'Montserrat-Bold',
  },
  colors: {
    ...
  },
};
```

**Comment by jaulz** [comment]
@solshuffle please check the new definition of fonts: https://github.com/callstack/react-native-paper/blob/master/src/styles/fonts.tsx

Instead of passing a string you need to pass an object with `fontFamily` and `fontWeight` keys.

**Comment by miladmeidanshahi** [comment]
> Hi @solshuffle
> 
> Which version of react-native-paper are you using ? Because all of you wrote should be already implemented in 3.0 branch
> 
> There is also padding="none" prop that allows to remove horizontal padding as you mentioned.
> 
> Please check the below screen from the example app:
> ![image](https://user-images.githubusercontent.com/21242757/62203897-5edf5a00-b38c-11e9-8fb0-8aa9ac3dfc47.png)

Does this prop work on version 2.16.0?

**Comment by aprilmintacpineda** [comment]
There is still no option to override the style for TextInput prop, I want to be able to set borderRadius, padding, specifying style on TextInput doesn't actually work as expected, unlike when using TextInput from React-Native.
